### DISCUSSION TITLE
Drawing into and out of a run straight

### DISCUSSION THREAD

**Discussion by coatsy35** [Courseplay/Courseplay_FS22#2358] [Ideas]
Would it be possible to have Courseplay wait until the implement has passed the end of the row before lilting out and also wait until the implement is level with the start of the row to lower it? Currently raise early/late and lift early/late do not do enough for trailed implements. In a real situation you would always draw straight into a new run, not start lowering an implement as you are turning into the run. Perhaps an  additional option to force the tractor and implement to draw into and out of the run straight, if the headland is big enough to support this? it's infuriating with big drills and trailed ploughs that the tractor does not drive out far enough into the headland so that it comes back in straight (even when working in lands - not that you'd do that for ploughing with a reversible plough) , even with large long mounted ploughs it misses that mark properly.

Also could courseplay not look at the ground contact point (or bar point as real GPS systems call it) so that it lifts and lowers when the correct point on the implement passes the mark? perhaps a setting on the course generation box with an adjuster on the hud? it could then put a horizontal line through the red box so we can see what point the implement will lift at. This point should also be used for driving around a curve i.e. when baling following a combine swath, a tractor should not be following the swath, the tractor should be driving so that the baler follows the swath (again just like in real life) otherwise you end up missing part of the swath on the curves like my pic below. 

![fsScreen_2023_02_26_10_58_02](https://user-images.githubusercontent.com/122838698/221411542-42245050-1d5d-418c-8966-faa58d8c8db6.png)

**Comment by Tensuko** [comment]
Lower and raise implement is controlled with a setting already.
Nothing will be changed there.
It is not comparable with RL to be able to control it.
The Tools have work areas and AI Markers that we use.

Baling in the curve is just natural, a dragged tool always drives the more inner line.
An Idea what we might implement at some time is a swath detection wich will correct for this problem.

Lower the tool more straight into the row, well try to use turn on field off if you have enough space.
It will not use reverse and might have the tool more straight.
And turn in pathfinder off.
Other then that, nothing is planed there and even then, would be low priority.

**Reply by coatsy35** [reply]
The raise and lower setting is not good enough though, it lowers implements when it shouldn't on certain tools. What position does this work from on the work area? from the front edge of it? could there by a forward rearward offset like the horizontal one? then the position for lowering could be altered

I have pathfinder off and turn on field activated. However if I have 10 headland runs with an implement width of 6m, the tractor and plough come out at the end of the swath and then turn inside of the first 3/4 headlands runs. It needs to turn further towards the worked land, out in a bigger "light bulb" shape and use more of the headland so that it it comes back into work straight. On the PW100 plough for instance by Giants, it needs long way to pull out with a large headland so it can turn properly and come back into work straight, otherwise the tractor starts turning into the run and then just lowers the plough whilst turning in. I see people mention on youtube that courseplay is not very good with trailed ploughs and it's not in my experience. Perhaps it is a limitation? is there more that can be done? 

In regarding driving around corners and swaths, this also affects ploughing as well, the tractor follows the line when the tractor should be driving wherever it needs to so the plough or baler follow the line. Could the steering not come from the centre of the work area?

Square corners like you said you might be able to do, so the corner is forced and the machine has to pull out and turn would help with the combining/mowing so the baler would do the same.

![Headlandloop](https://user-images.githubusercontent.com/122838698/221421949-57ece54d-eb2f-44ca-ae50-a599e02bb1bd.png)

**Reply by Tensuko** [reply]
About round corners: It is normal and won't be changed by us as this is simple physic!
![round pattern](https://user-images.githubusercontent.com/27344744/221422596-35e5fa5a-e606-4f2e-a9e2-27d55c6ba61c.png)
![round corners](https://user-images.githubusercontent.com/27344744/221422592-adb37b1b-2b50-4c2a-bd4e-6366212bf394.png)

That setting is all that can be done, there is no "good enough" it is either this or that.
![lowerraise](https://user-images.githubusercontent.com/27344744/221422845-c5fad7b0-daff-447a-a9db-909d9d0d1db7.png)
The work area is in the model itself and got nothing to do with a visual "tool" that we see. The better Giants placed it, the better it looks.
So lower late is we lower the tool with the end of the work area wich might be too late for the row.
Lower early is lower it with the start of the work area.

"I have pathfinder off and turn on field activated"
That why I said turn on field off. Try it. Other then that, we won't change anything there any time soon.

Overall I don't see you problem with that lowering and raising....

**Comment by coatsy35** [comment]
Thank you for the explanation regarding the work area now I understand.

I see what the 60 year old book is saying. very inefficient these days and you won't find many farmers practising that unless they are in very large fields or the turn radius is such that a trailed implement will turn the corner nicely and match up. Most would pull out into the corners, stop at the wall or hedge and then back up turn and pull back into the other corner or re-angle themselves and pull out again. Real GPS systems struggled with this for years, now most have a setting called square headlands as I've mentioned before, which essentially projects a virtual line out of the field so the guidance pulls them into a corner straight, this would be on a setting that says any corner over so many degrees turn into a square corner whilst leaving nice natural curves where the implement will match up as part of the headland pattern.

I will record what I'm seeing regards to raising and lowering, I guess what I am seeing is the result of an implement not pulling out far enough on the headland and coming back in straight so it matches up. I will try with turn in field turned off.

With regards to turning a curved corner with something like a baler, I'm curious is it possible to make the tractor steer so the implement work area stays on the line rather than the tractor thus keeping a baler on swath or help a drill around it trickier corner.

Thank you for your time.

**Reply by Tensuko** [reply]
"I will record what I'm seeing regards to raising and lowering, I guess what I am seeing is the result of an implement not pulling out far enough on the headland and coming back in straight so it matches up. I will try with turn in field turned off."

Not pulling out far enough -> change raise implement to late.

**Reply by coatsy35** [reply]
I already have it set to late always do. and lower early.
